### DISCUSSION TITLE
Fallowing logoutURL does not remove the session cookie.

### DISCUSSION THREAD

**Discussion by atstoyanov** [ory/kratos#1582] [Q&A] [answered]
Hello guys,

I'm trying v0.7 and really love the AJAX flows. However I have an issue with the logout. Fallowing the generated logout URL doesn't invalidate the ory_krato_session cookie which makes problem with our server configuration. We are using Oathkeepr to protect our GraphQL API. We relay on the `ory_kratos_session` cookie. If the cookie is not present the requests are redirected directly to the graphql. For the graphql the requests are looking like comming from unauthenticated user and public queries and mutations are available to use. When the cookie is present the cookie is validated and transformed into a JWT token which  the GraphQL's guards can understand. Unfortunately after the logout the Oathkeeper start returning 401 because the session is invalidated on the server, but the session cookie is still present in the request. The request is blocked by Oathkeeper and the graphql api is not accessible by the unauthenticated user.

Is not invalidating the cookie on the client intentional or a bug?

Using the 0.6 the logout was working perfect and the cookie was invalidated properly.

Also from the documentation is not clear how the redirect to the post logout return URL happens.

P.S. I understand that in the future if we are able to invalidate the session from admin panel, we will finish with the same situation.

**Comment by aeneasr** [comment]
How do you invalidate it? So exact REST requests :)

**Reply by atstoyanov** [reply]
Navigating to `/self-service/logout?token=<token>`. 

The actual URL is generated from calling `kratosClient.createSelfServiceLogoutFlowUrlForBrowsers()`

**Reply by aeneasr** [reply]
Thank you, possibly a bug. Any trace or fix is appreciated!

**Comment by dibyajyotibehera** [comment, marked as answer]
i tried the spa flow looks to me like the cookie is getting deleted.
created a video here . - https://www.loom.com/share/aa49b6e53b0146b1a6ff8ca8861b7dc5

anything missing here ?

**Reply by aeneasr** [reply]
Yes, this was recently fixed iirc :)

**Reply by kezhang-wish** [reply]
Do you have the commit ID for fix? We are currently using 0.7.1 alpha

**Reply by kezhang-wish** [reply]
@aeneasr could you please test it again on self-service ui, it doesn't seem to be removing the session cookie from browser after logout, even though the token has been invalidated.

**Reply by aeneasr** [reply]
version?

**Reply by kezhang-wish** [reply]
0.7.1 alpha

**Comment by kezhang-wish** [comment]
After testing, seems like the cookie is not being cleared, using the browser flow.

**Comment by edardev** [comment]
True, it is not deleting the session on Postgres after login from the client and hitting the logout route on http://127.0.0.1:4433/self-service/logout?token=logout_token

I used the following docker images and also grab the logout_token and made a request to the logout from above.
image: oryd/kratos:v0.8.0-alpha.3
image: oryd/kratos-selfservice-ui-node:v0.8.0-alpha.3
image: postgres:9.6

**Reply by aeneasr** [reply]
CAn you pleaes create an issue with an easy to follow reproducible flow? :)
